packages/datatrak-web/src/features/Questions/TextQuestion.tsx

```diff
@@ -0,0 +1,51 @@
+/*
+ * Tupaia
+ *  Copyright (c) 2017 - 2023 Beyond Essential Systems Pty Ltd
+ */
+
+import React from 'react';
+import { FormControlLabel, TextField } from '@material-ui/core';
+import { SurveyQuestionInputProps } from '../../types';
+import styled from 'styled-components';
+
+const Label = styled(FormControlLabel)`
+  width: calc(100% - 3.5rem);
+  align-items: flex-start;
+  margin: 0;
+  .MuiInput-root {
+    font-size: 0.875rem;
+  }
+`;
+
+enum FIELD_TYPES {
+  FreeText = 'text',
+  Number = 'number',
+}
+
+export const TextQuestion = ({
+  id,
+  label,
+  name,
+  inputRef,
+  type,
+  controllerProps,
+}: SurveyQuestionInputProps) => {
+  return (
+    <Label
+      label={label}
+      name={name!}
+      labelPlacement={'top'}
+      inputRef={inputRef}
+      {...controllerProps}
```

Comment:
It's probably a pretty annoying request but can we explicitly add the controller props. It's something that we have discussed previously in dev chat and there was a consensus that it's more readable. There should only be four (onChange, onBlur, value, name, ref)

Reply:
Also I think it would be ideal to set the form state using the onBlur event for text inputs. eg. onBlur={onChange} so that we are not setting state unnecessarily on each keypress.

Reply:
I can definitely explicitly add them. As for making it onBlur, because it's a controlled input, if we make it on blur, then the value the user types won't show up in the field until it's unfocussed, which I guess we don't want, right?